### ISSUE TITLE
otel status value for the graphql engine metadata cannot be sourced from env vars similar to other otel properties

### ISSUE THREAD

**Issue by krisiye** [hasura/graphql-engine#10177]
### Version Information

Server Version:
CLI Version (for CLI related issue):
`v2.37.0`

### Environment

All environments. Self Hosted and Cloud.

### What is the current behaviour?

Hasura Metadata at the moment does not allow one to turn off open telemetry on a given stage if you chose to. The typical use-case is to allow some flexibility to turn on/off local or non prod stages.

### What is the expected behaviour?

1. Expected behavior is to allow otel status to be read from an environment var similar to any other OTEL config.

### How to reproduce the issue?

Try making otel status a variable that we expect to read from env.

### Screenshots or Screencast

NA
### Please provide any traces or logs that could help here.

Appears to be not compatible with {{otel.status}} (assuming an env var `otel.status` injected with this name):
```error: yaml: invalid map key: map[string]interface {}{"OTEL_STATUS":interface {}(nil)} ```

Also tried to put vars behind quotes and got the following:
```FATA[0001] error applying metadata 
{
  "error": "OpenTelemetry status must be either \"enabled\" or \"disabled\"",
  "path": "$.args.metadata.opentelemetry.status",
  "code": "parse-failed"
} ```


### Any possible solutions/workarounds you're aware of?

NA

### Keywords

open telemetry

**Comment by rikinsk** [comment]
FIxed via https://github.com/hasura/graphql-engine/commit/ce5ff4c367f793400b594fdde7b12388db93990f

**Comment by krisiye** [comment]
@rikinsk  - which engine version would include the fix?

**Comment by manasag** [comment]
This is available in v2.39 released last week.
